src/core/magnolia.scala

```diff
@@ -77,12 +79,27 @@ trait Derivation[TypeClass[_]]
       case sum: Mirror.SumOf[A]         => derivedMirrorSum[A](sum)
       case product: Mirror.ProductOf[A] => derivedMirrorProduct[A](product)
 
-  inline def derived[A](using Mirror.Of[A]): Typeclass[A] = derivedMirror[A]
+  inline def derivedValueClass[A]: Typeclass[A] = join(CaseClassDerivation.valueClassDerivation)
+
+  inline def derivedMirrorless[A]: Typeclass[A] = 
+    inline if Macro.isValueClass[A] then 
+      derivedValueClass[A]
+    else error("Mirrrorful derivation not posiible, mirrorless derivation either not possible or not implemented yet.")
```

Comment:
typos :) plus I would make this a bit more user-friendly, sth like `Deriving the typeclass based on mirrors or directly is not possible for $A. Please refer to the documentation or report a feature request` (it's important to include the type name as this can be a nested invocation)

Reply:
Refactored.